File: src/Application Specific/Classes/Entity Collections/Order/PartSupplierPriceRequest/PartSupplierPriceRequestValidator.vb
Namespace Entity

Namespace PartSupplierPriceRequests

Public Class PartSupplierPriceRequestValidator
    'make sure that contact object is valid
    Inherits BaseValidator

    Public Sub Validate(ByVal oPartSupplierPriceRequest As PartSupplierPriceRequest)

        'make sure that contact object is valid
        If oPartSupplierPriceRequest.Errors.Count > 0 Then
            Dim de As DictionaryEntry
            For Each de In oPartSupplierPriceRequest.Errors
                Me.AddCriticalMessage(CStr(de.Value))
            Next
        End If

                If oPartSupplierPriceRequest.OrderID = 0 Then
                    Me.AddCriticalMessage("Order missing")
                End If
                If oPartSupplierPriceRequest.SupplierID = 0 Then
                    Me.AddCriticalMessage("Supplier missing")
                End If
                If oPartSupplierPriceRequest.PartID = 0 Then
                    Me.AddCriticalMessage("Part missing")
                End If
                If Not oPartSupplierPriceRequest.QuantityInPack > 0 Then
                    Me.AddCriticalMessage("Quantity must be greater than 0")
                End If

                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

    End Sub

End Class

End Namespace

End Namespace

File: apollo/apollo.fsm/src/Application Specific/Classes/Entity Collections/Order/PartSupplierPriceRequest/PartSupplierPriceRequestValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace PartSupplierPriceRequests
    {
        // make sure that contact object is valid
        public class PartSupplierPriceRequestValidator : BaseValidator
        {
            public void Validate(PartSupplierPriceRequest oPartSupplierPriceRequest)
            {
                // make sure that contact object is valid
                if (oPartSupplierPriceRequest.Errors.Count > 0)
                {
                    foreach (DictionaryEntry de in oPartSupplierPriceRequest.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (oPartSupplierPriceRequest.OrderID == 0)
                {
                    AddCriticalMessage("Order missing");
                }

                if (oPartSupplierPriceRequest.SupplierID == 0)
                {
                    AddCriticalMessage("Supplier missing");
                }

                if (oPartSupplierPriceRequest.PartID == 0)
                {
                    AddCriticalMessage("Part missing");
                }

                if (!(oPartSupplierPriceRequest.QuantityInPack > 0))
                {
                    AddCriticalMessage("Quantity must be greater than 0");
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}